archive/progress_common/src/main/scala/uk/ac/wellcome/platform/archive/common/progress/monitor/ProgressTracker.scala

```diff
@@ -12,18 +12,19 @@ import grizzled.slf4j.Logging
 import uk.ac.wellcome.platform.archive.common.progress.models._
 import uk.ac.wellcome.storage.dynamo._
 
+import scala.concurrent.{ExecutionContext, Future, blocking}
 import scala.util.{Failure, Success, Try}
 
 class ProgressTracker @Inject()(
   dynamoClient: AmazonDynamoDB,
   dynamoConfig: DynamoConfig
-) extends Logging {
+)(implicit ex: ExecutionContext) extends Logging {
   import Progress._
 
-  def get(id: UUID): Option[Progress] = {
-    Scanamo.get[Progress](dynamoClient)(dynamoConfig.table)(
+  def get(id: UUID): Future[Option[Progress]] = Future {
+    blocking(Scanamo.get[Progress](dynamoClient)(dynamoConfig.table)(
```

Comment:
Why `blocking`?

Reply:
https://www.cakesolutions.net/teamblogs/demystifying-the-blocking-construct-in-scala-futures
Basically, it's to tell the execution context that this operation is going to block, so that the execution context can treat it differently. Not all execution contexts do something special for blocking operations but it's good practice to add it, so the ones that do can be notified